source/postprocess/visualization/named_additional_outputs.cc

```diff
@@ -0,0 +1,171 @@
+/*
+  Copyright (C) 2011 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/named_additional_outputs.h>
+#include <aspect/material_model/interface.h>
+#include <aspect/utilities.h>
+
+#include <algorithm>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      NamedAdditionalOutputs<dim>::
+      NamedAdditionalOutputs ()
+        :
+        DataPostprocessor<dim> ()
+      {}
+
+      template <int dim>
+      std::vector<std::string>
+      NamedAdditionalOutputs<dim>::
+      get_names () const
+      {
+        static std::vector<std::string> property_names;
```

Comment:
I can not fill it in the constructor, because we need SimulatorAccess and we don't have that yet in the constructor. Instead, I added an initialize function to the visualization postprocessor and used that. 
Usually I would have made a separate pull request for that, but then I thought it would be good to have something that tests if the initialize function works, so I added it to this one.